### PR TITLE
Authentication support

### PR DESCRIPTION
**Pull request by bulusoy**
It came to my attention that PINRemoteImage does not feature authentication handling. Since I need it, I just implemented it :-)

### CODE REVIEW

**Comment by bulusoy** [comment]
Failing is really weird. I'm able to build and run it locally either from IDE or command-line. @garrettmoon @jparise can you take a look into this please?

**Comment by jparise** [comment]
@bulusoy I think the CI build failed because `__nullable` isn't recognized by older compilers / SDKs.

File: Pod/Classes/PINRemoteImageManager.m [hunk 1]
```diff
@@ -113,6 +113,7 @@ @interface PINRemoteImageManager () <PINURLSessionManagerDelegate>
 @property (nonatomic, assign) float highQualityBPSThreshold;
 @property (nonatomic, assign) float lowQualityBPSThreshold;
 @property (nonatomic, assign) BOOL shouldUpgradeLowQualityImages;
+@property (nonatomic, strong) PINRemoteImageManagerAuthenticationChallenge authenticationChallenge;
```
Comment (jparise) [inline Pod/Classes/PINRemoteImageManager.m]: I usually `copy` block properties.

**Comment by garrettmoon** [comment]
Can you write tests for this support? I know they can be onerous, however they really help with detecting regressions!

**Comment by garrettmoon** [comment]
Oh, and thank you for this! I knew it would be something we needed to add at some point :)

**Comment by bulusoy** [comment]
My pleasure pal, ok I'll write tests, no problem. I'll attach them to this PR

**Comment by garrettmoon** [comment]
Awesome, thank you!

File: Pod/Classes/PINRemoteImageManager.m (hunk 1 above)
Comment (bulusoy) [inline Pod/Classes/PINRemoteImageManager.m]: Ok, will also change hat :)

**Comment by bulusoy** [comment]
@garrettmoon  added tests, but for some reason travis is still complaining

**Comment by garrettmoon** [comment]
Here's the error:

```
In file included from /Users/travis/build/pinterest/PINRemoteImage/Pod/Classes/Image Categories/FLAnimatedImageView+PINRemoteImage.m:9:
In file included from /Users/travis/build/pinterest/PINRemoteImage/Pod/Classes/Image Categories/FLAnimatedImageView+PINRemoteImage.h:11:
/Users/travis/build/pinterest/PINRemoteImage/Pod/Classes/PINRemoteImageManager.h:103:157: error: expected ')'
typedef void(^PINRemoteImageManagerAuthenticationChallengeCompletionHandler)(NSURLSessionAuthChallengeDisposition disposition, NSURLCredential * __nullable credential);
                                                                                                                                                            ^
/Users/travis/build/pinterest/PINRemoteImage/Pod/Classes/PINRemoteImageManager.h:103:77: note: to match this '('
typedef void(^PINRemoteImageManagerAuthenticationChallengeCompletionHandler)(NSURLSessionAuthChallengeDisposition disposition, NSURLCredential * __nullable credential);
```

I think you missed removing some nullables?

**Comment by bulusoy** [comment]
@garrettmoon  Yes I did. Mia culpa.

**Comment by garrettmoon** [comment]
Sweet!

**Pull request merged**